lis/surfacemodels/land/ac.7.2/ac_modules/utils.f90

```diff
@@ -0,0 +1,335 @@
+
+    str = '7.2'
+end function GetVersionString
+
+
+function int2str(num) result(str)
+    !! Converts an integer into a string.
+    integer, intent(in) :: num
+    character(len=:), allocatable :: str
+
+    character(len=128) :: tmp
+
+    write(tmp, '(i0)') num
+    str = trim(tmp)
+end function int2str
+
+
+function roundc_int32(x, mold) result(y)
+    !! Returns commercial rounds, following Pascal's banker's rules for rounding
+    real(sp), intent(in) :: x
+        !! Value to be rounded to an integer
+    integer(int32), intent(in) :: mold
+        !! Integer determining the kind of the integer result
+    integer(int32) :: y
+    real(sp) :: x_clipped
+
+    ! Check if x is within the tolerated range for int32 before rounding
+    if (x > 2147483647._sp) then
+        x_clipped = 2147483647._sp
+    elseif (x < -2147483648._sp) then
+        x_clipped = -2147483648._sp
+    else
+        x_clipped = x
+    end if
+
+    ! Rounding logic based on x_clipped
+    if (abs(x_clipped - floor(x_clipped, kind=int32) - 0.5_sp) < epsilon(0._sp)) then
+       if (x_clipped > 0) then
+          if (mod(abs(trunc(x_clipped)),2) == 0) then
+              y = floor(x_clipped, kind=int32)
+          else
+              y = ceiling(x_clipped, kind=int32)
+          end if
+       else
+          if (mod(abs(trunc(x_clipped)),2) == 0) then
+              y = ceiling(x_clipped, kind=int32)
+          else
+              y = floor(x_clipped, kind=int32)
+          end if
+       end if
+    else
+       ! Standard round for values not ending on 0.5
+       y = nint(x_clipped, kind=int32)
+    end if
+
+end function roundc_int32
+
+
+
+function roundc_int8(x, mold) result(y)
+    !! Returns commercial rounds, following Pascal's banker's rules for rounding
+    real(sp), intent(in) :: x
+        !! Value to be rounded to an integer
+    integer(int8), intent(in) :: mold
+        !! Integer determining the kind of the integer result
+    integer(int8) :: y
+    real(sp) :: x_clipped
+
+    ! Check if x is within the tolerated range for int8 before rounding
+    if (x > 127._sp) then
+        x_clipped = 127._sp
+    elseif (x < -128._sp) then
+        x_clipped = -128._sp
+    else
+        x_clipped = x
+    end if
+
+    ! Rounding logic based on x_clipped
+    if (abs(x_clipped - floor(x_clipped, kind=int32) - 0.5_sp) < epsilon(0._sp)) then
+       if (x_clipped > 0) then
+          if (mod(abs(trunc(x_clipped)),2) == 0) then
+             y = floor(x_clipped, kind=int8)
+          else
+             y = ceiling(x_clipped, kind=int8)
+          end if
+       else
+          if (mod(abs(trunc(x_clipped)),2) == 0) then
+             y = ceiling(x_clipped, kind=int8)
+          else
+             y = floor(x_clipped, kind=int8)
+          end if
+       end if
+    else
+       ! Standard round for values not ending on 0.5
+       y = nint(x_clipped, kind=int8)
+    end if
+
+end function roundc_int8
+
+
+function trunc(x) result(y)
```

Comment:
This function is also in ac_utils? Not sure what is where why.

Reply:
Yes, this function is in ac_utils under ldt/ because it is needed to compute the compartment size. 
Since LIS and LDT are two separate executables, the utils are needed twice.